datastore/sample.rb

```diff
@@ -0,0 +1,748 @@
+  end
+  # [END entity_with_parent]
+
+  task
+end
+
+def properties
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START properties]
+  task = datastore.entity "Task" do |t|
+    t["type"] = "Personal"
+    t["created"] = Time.now
+    t["done"] = true
+    t["priority"] = 4
+    t["percent_complete"] = 10.0
+    t["description"] = "Learn Cloud Datastore"
+  end
+  # [END properties]
+end
+
+def array_value
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START array_value]
+  task = datastore.entity "Task", "sampleTask" do |t|
+    t["tags"] = %w(fun programming)
+    t["collaborators"] = %w(alice bob)
+  end
+  # [END array_value]
+end
+
+def basic_entity
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START basic_entity]
+  task = datastore.entity "Task" do |t|
+    t["type"] = "Personal"
+    t["done"] = true
+    t["priority"] = 4
+    t["description"] = "Learn Cloud Datastore"
+  end
+  # [END basic_entity]
+
+  task
+end
+
+def upsert
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START upsert]
+  task = datastore.entity "Task", "sampleTask" do |t|
+    t["type"] = "Personal"
+    t["done"] = true
+    t["priority"] = 4
+    t["description"] = "Learn Cloud Datastore"
+  end
+  datastore.save task
+  # [END upsert]
+
+  task
+end
+
+def insert
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START insert]
+  task = datastore.entity "Task" do |t|
+    t["type"] = "Personal"
+    t["done"] = true
+    t["priority"] = 4
+    t["description"] = "Learn Cloud Datastore"
+  end
+  datastore.save task
+  # [END insert]
+
+  task
+end
+
+def lookup
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START lookup]
+  task_key = datastore.key "Task", "sampleTask"
+  task = datastore.find task_key
+  # [END lookup]
+
+  task
+end
+
+def update
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  task = datastore.entity "Task", "sampleTask" do |t|
+    t["type"] = "Personal"
+    t["done"] = true
+    t["priority"] = 4
+    t["description"] = "Learn Cloud Datastore"
+  end
+  datastore.save task
+
+  # [START update]
+  task = datastore.find "Task", "sampleTask"
+  task["priority"] = 5
+  datastore.save task
+  # [END update]
+
+  task
+end
+
+def delete
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START delete]
+  task_key = datastore.key "Task", "sampleTask"
+  datastore.delete task_key
+  # [END delete]
+
+  task_key
+end
+
+def batch_upsert
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START batch_upsert]
+  task1 = datastore.entity "Task" do |t|
+    t["type"] = "Personal"
+    t["done"] = true
+    t["priority"] = 4
+    t["description"] = "Learn Cloud Datastore"
+  end
+
+  task2 = datastore.entity "Task" do |t|
+    t["type"] = "Personal"
+    t["done"] = true
+    t["priority"] = 5
+    t["description"] = "Integrate Cloud Datastore"
+  end
+
+  task_key1, task_key2 = datastore.save(task1, task2).map(&:key)
```

Comment:
Instead of using a `Symbol#to_proc` `map` in the sample, how about making this slightly more verbose but more explicit?

```ruby
#...
task2 = datastore.entity "Task" do |t|
  t["type"] = "Personal"
  t["done"] = true
  t["priority"] = 5
  t["description"] = "Integrate Cloud Datastore"
end

entities = datastore.save task1, task2

task_key1 = entities[0].key
task_key2 = entities[1].key
```

Inspired by the Java Batch Operations sample:
https://cloud.google.com/datastore/docs/concepts/entities#batch_operations